packages/frontend/src/components/common/PivotTable/index.tsx

```diff
@@ -92,24 +160,250 @@ const PivotTable: FC<PivotTableProps> = ({
         [data.rowTotalFields, getField],
     );
 
-    const getMetricAsRowTotalValueFromAxis = useCallback(
-        (total: unknown, rowIndex: number): ResultValue | null => {
-            const value = last(data.indexValues[rowIndex]);
-            if (!value || !value.fieldId) throw new Error('Invalid pivot data');
-
-            const item = getField(value.fieldId);
-            if (!isSummable(item)) {
-                return null;
+    // TODO: indexValues, dataValues, and rowTotalFields should perhaps be combined in pivotQueryResults.ts
+    //  or elsewhere but doing it here for now.
+    //  Alternatively, this whole pivot table implementation should perhaps be combined with the non-pivot tables.
```

Comment:
Yes, I think we should move this logic to `pivotQueryResults.ts` so all the data is calculated in that function and this component is only about rendering and actions.
If the outcome of a hook are values and those values are not updated anywhere else, it makes sense to be a `useMemo` and avoid multiple useState.

Reply:
- Moved data structure "retrofit" to pivotQueryResults.ts
- Changed useState to useMemo where applicable
OK, I was thinking that pivotQueryResults.ts could be reworked to remove the "retrofit" when pivot and non-pivot components are combined.